src/librustc/ty/context.rs

```diff
@@ -307,7 +307,8 @@ pub struct ResolvedOpaqueTy<'tcx> {
 ///
 /// Here, we would store the type `T`, the span of the value `x`, and the "scope-span" for
 /// the scope that contains `x`.
-#[derive(RustcEncodable, RustcDecodable, Clone, Debug, Eq, Hash, HashStable, PartialEq)]
+#[derive(RustcEncodable, RustcDecodable, Clone, Debug, Eq, Hash, HashStable, PartialEq,
+    TypeFoldable)]
```

Comment:
Maybe we should have two `#[derive(...)]` attributes, one for the standard derives and one for the rustc-specific derives?